JoinColony/colonyCDapp
# Fix: Status of fund expenditure motion stake being always "Staking"

## Description

While reviewing #2772, I came across a bug where status of stakes made on fund expenditure motions (and any other expenditure related motions) would always get overwritten to `Staking`. This PR fixes the problem by checking the type of the stake when determining the status.

## Testing

To replicate the original issue:
- Checkout `feat/advanced-payments-ui`.
- Install voting reputation extension.
- Create an advanced payment. Fund it via a motion.
- Fully stake the motion. Run `npm run forward-time 1` to pass the staking period.
- Check the stake status in the user hub. It shows as `Staking` instead of `Finalizable`: 
<img width="703" alt="image" src="https://github.com/user-attachments/assets/0afb59d7-4e95-42d0-88e4-9f2a593c9bc0">


On this branch, the status should show correctly:
<img width="696" alt="image" src="https://github.com/user-attachments/assets/f908392c-def3-416f-83d1-7e1bcd8a710d">
Once you finalize the motion, the status should update to Claimable.
